astro/src/content/docs/extend/events-and-webhooks/events/user-login-failed.mdx

```diff
@@ -47,6 +47,22 @@ export const eventType = 'user.login.failed';
     Moved to <InlineField>event.info</InlineField> in 1.30.0
   </APIField>
 
+  <APIField slot="trailing-fields" name="event.reason" type="Object" since="1.53.0">
+    An object containing data on the reason the login failed if it was prevented by a [Login Validation Lambda](/docs/extend/code/lambdas/login-validation).
+  </APIField>
+
+  <APIField slot="trailing-fields" name="event.reason.code" type="String" since="1.53.0">
+    The theme message key used to find a message to display to the end user.
+  </APIField>
+
+  <APIField slot="trailing-fields" name="event.reason.lambdaId" type="UUID" since="1.53.0">
+    The lambda Id which triggered the login failure.
+  </APIField>
+
+  <APIField slot="trailing-fields" name="event.reason.lambdaResult" type="Errors" since="1.53.0">
+    The [Errors](/docs/apis/errors) displayed to the end user. 
```

Comment:
This is only present in the event if the `reason.code` is `lambdaValidation`.  These are not necessarily displayed to the user. All we know for sure is that this is the error object returned by the lambda function that caused the login to fail. 

We have no idea if the API caller showed the user these errors, or if they are using a theme, if these messages were localized or modified to something else.